packages/webapp/src/pages/election/round-video-upload.tsx

```diff
@@ -24,19 +22,21 @@ import {
     VideoSubmissionFormAndPreview,
     VideoSubmissionPhase,
 } from "_app/ui";
+
 import {
     CurrentElection,
     ElectionStatus,
     ErrorLoadingElection,
     setElectionRoundVideo,
-    getRoundTimes,
 } from "elections";
-import { EdenMember } from "members";
 import { RoundHeader } from "elections/components/ongoing-election-components";
+import { MemberAccountData } from "members";
 
 export const RoundVideoUploadPage = () => {
     const [isElectionComplete, setIsElectionComplete] = useState(false);
-    const [uploadLimitTime, setUploadLimitTime] = useState(dayjs());
+    const [uploadLimitTime, setUploadLimitTime] = useState(
+        dayjs().add(1, "day") // set default limit in the future while election data is being loaded
+    );
 
     const {
         data: currentElection,
```

Comment:
Re: Line 36 - line 65:

suggestion: I think this is a case where we can reactively derive state instead of setting state. Could we get rid of both `useState` and `useEffect` calls here with something like this instead:

```
    const {
        data: currentElection,
        isLoading: isLoadingElection,
        isError: isErrorElection,
    } = useCurrentElection();
    const { data: electionState } = useElectionState();
    
    let uploadTimeLimit = electionState
        ? dayjs(electionState.last_election_time + "Z").add(48, "hour")
        : dayjs().add(1, "day");

    const isAfterUploadLimitTime =
        dayjs().isAfter(uploadTimeLimit) &&
        currentElection?.electionState === ElectionStatus.Registration;
```

Then we in the JSX on `isAfterUploadLimitTime`.

Reply:
Also, should we react to `isLoading` and `isError` from `useElectionState()` like we do from `useCurrentElection()`?

Reply:
yes, done!